### PR TITLE
feat(widget-chat): Add scrollable component wrapper

### PR DESCRIPTION
**Pull request by bzang**

### CODE REVIEW

File: packages/widget-chat/src/components/wrapper-scrollable/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react';
+
+import styles from './styles.css';
+
+function getDisplayName(C) {
+  return C.displayName || C.name || `C`;
+}
+
+export default function scrollableComponent(WrappedComponent) {
```
Comment (ianwremmel) [inline packages/widget-chat/src/components/wrapper-scrollable/index.js]: I feel like your function name and HOC name are backwards...

File: packages/widget-chat/src/components/wrapper-scrollable/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  class InjectScrollable extends Component {
+    constructor(props) {
+      super(props);
+      this.getNode = this.getNode.bind(this);
```
Comment (ianwremmel) [inline packages/widget-chat/src/components/wrapper-scrollable/index.js]: maybe take a look at the autobind decorator for this.

File: packages/widget-chat/src/containers/chat-widget/index.js [hunk 3]
```diff
@@ -52,6 +57,12 @@ export class ChatWidget extends Component {
     return nextProps.sparkState.connected !== props.sparkState.connected || nextProps.user !== props.user || nextProps.conversation !== props.conversation;
   }
 
+  componentDidUpdate() {
+    if (this.activityList) {
```
Comment (ianwremmel) [inline packages/widget-chat/src/containers/chat-widget/index.js]: So, here's where it gets hairy. you only want to scroll to the bottom if the message was sent from this device
or the viewport was already at the bottom when the message came in

Comment (bzang) [inline packages/widget-chat/src/containers/chat-widget/index.js]: Got it, i'll add the logic.

File: packages/widget-chat/src/components/wrapper-scrollable/index.js (hunk 1 above)
Comment (bzang) [inline packages/widget-chat/src/components/wrapper-scrollable/index.js]: Haha good call.

File: packages/widget-chat/src/components/wrapper-scrollable/index.js (hunk 2 above)
Comment (ianwremmel) [inline packages/widget-chat/src/components/wrapper-scrollable/index.js]: yep, that's the one. i'm pretty much always for adding dependencies over building it ourselves.

File: packages/widget-chat/src/components/scroll-to-bottom-button/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export default function ScrollToBottomButton(props) {
+  const {
+    onClick,
+    label
+  } = props;
+
+  const buttonStyle = label ? styles.withText : ``;
+  return (
+    <div className={classNames(`scroll-to-bottom-container`, styles.container)} onClick={onClick} >
```
Comment (adamweeks) [inline packages/widget-chat/src/components/scroll-to-bottom-button/index.js]: Do we want the click on the div or on the button?

File: packages/widget-chat/src/containers/chat-widget/index.js [hunk 5]
```diff
@@ -19,6 +28,13 @@ import injectSpark from '../../modules/redux-spark/inject-spark';
  */
 export class ChatWidget extends Component {
 
+  constructor(props) {
+    super(props);
+    this.getActivityList = this.getActivityList.bind(this);
+    this.handleScrollToBottom = this.handleScrollToBottom.bind(this);
+    this.handleScroll = _.debounce(this.handleScroll.bind(this), 150);
```
Comment (bzang) [inline packages/widget-chat/src/containers/chat-widget/index.js]: Yea. That's how I had it before but the function inside would never fire. Not sure why that was the case and didn't have time to pursue. Any thoughts @ianwremmal?

**Pull request merged**